File: 4673/4673.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    4673.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2021/09/27 14:38:36 by jihoolee          #+#    #+#              #
#    Updated: 2021/09/27 14:38:37 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def get_selfnums_lst(num):
	result = []
	for i in range(1, num):
		temp = i + sum(map(int, list(str(i))))
		if (temp <= num):
			result.append(temp)
	return (result)

selfnums_lst = get_selfnums_lst(10000)

for i in range(1, 10000 + 1):
	if (i not in selfnums_lst and i <= selfnums_lst[-1]):
		print(i)

File: 4673/4673.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   4673.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2021/09/27 14:38:19 by jihoolee          #+#    #+#             */
/*   Updated: 2021/09/27 14:38:30 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <vector>

using namespace std;

bool	is_selfnum(int i)
{
	int	sum;
	int	temp;

	for (int j = 1; j < i; j++)
	{
		sum = j;
		temp = j;
		while (temp)
		{
			sum += temp % 10;
			temp /= 10;
		}
		if (sum == i)
			return (false);
	}
	return (true);
}

int	main(void)
{
	for (int i = 1; i <= 10000; i++)
		if (is_selfnum(i) == true)
			cout << i << endl;
	return (0);
}
